### PR TITLE
[#53] (Proposal 1) Removed IntegrationPointAppyxProvider

### LINKED ISSUE #53: Interop Integrationpoint shouldn't force to expose appyx interface
Current interop integration point required activities to implement `IntegrationPointAppyxProvider`.

This expose the interface in module edges, forcing parent modules to implement appyx event if they are not using it.

Example.

`Module A` -> `Module B`

`Module B` implements `SomeActiviy: RibAcitivty(), IntegrationPointAppyxProvider{...}`
`Module A` has no appyx dependencies.

Compilation failed as Module A will try to resolve  `IntegrationPointAppyxProvider`

### PR DESCRIPTION
**Pull request by LachlanMcKee**
## Description
Possible solution for avoiding exposing the appyx dependencies in modules that reference an Activity that extends IntegrationPointAppyxProvider

Fixes #53 

## Check list

- [x] I have updated `CHANGELOG.md` if required.
- [x] I have updated documentation if required.

### CODE REVIEW

File: core/src/main/kotlin/com/bumble/appyx/core/integrationpoint/ActivityIntegrationPoint.kt [hunk 1]
```diff
@@ -44,4 +48,44 @@ open class ActivityIntegrationPoint(
...
     }
+
+    companion object {
+        private val integrationPoints = WeakHashMap<AppCompatActivity, IntegrationPoint>()
+
+        fun createIntegrationPoint(
+            activity: AppCompatActivity,
+            savedInstanceState: Bundle?,
+        ): ActivityIntegrationPoint =
+            ActivityIntegrationPoint(activity, savedInstanceState)
```
Comment (KovalevAndrey) [inline core/src/main/kotlin/com/bumble/appyx/core/integrationpoint/ActivityIntegrationPoint.kt:59]: shouldn't we check for existing integration point for this activity before creating it?

**Review by KovalevAndrey — APPROVED** [review]

File: core/src/main/kotlin/com/bumble/appyx/core/integrationpoint/ActivityIntegrationPoint.kt [hunk 2]
```diff
@@ -44,4 +48,44 @@ open class ActivityIntegrationPoint(
             activity.finish()
         }
     }
+
+    companion object {
+        private val integrationPoints = WeakHashMap<AppCompatActivity, IntegrationPoint>()
+
+        fun createIntegrationPoint(
+            activity: AppCompatActivity,
```
Comment (KovalevAndrey) [inline core/src/main/kotlin/com/bumble/appyx/core/integrationpoint/ActivityIntegrationPoint.kt:56]: let's align with https://github.com/bumble-tech/appyx/pull/132/files#diff-61d49596405e9cbcac4b41945813a8097a38cb5adecb8b0826061acc93fc3fe4R12

Comment (LachlanMcKee) [reply core/src/main/kotlin/com/bumble/appyx/core/integrationpoint/ActivityIntegrationPoint.kt:56]: @KovalevAndrey done

File: core/src/main/kotlin/com/bumble/appyx/core/integrationpoint/ActivityIntegrationPoint.kt (hunk 1 above)
Comment (LachlanMcKee) [reply core/src/main/kotlin/com/bumble/appyx/core/integrationpoint/ActivityIntegrationPoint.kt:58]: The existing API doesn't work this way. We could do a getOrCreate or something similar, but I wanted to remain reasonably consistent.

**Review by KovalevAndrey — APPROVED** [review]

**Pull request merged**